third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/ConvertLayoutOpToLLVM/SharedToDotOperandMMAv2OrV3.cpp

```diff
@@ -408,8 +408,10 @@ MMA16816SmemLoader::loadX4(int batch, int mat0, int mat1, ArrayRef<Value> ptrs,
     int canonWidth = (8 * elemBytes * inc) / canonBits;
     Type canonInt = int_ty(canonBits);
     std::array<Value, 4> retElems;
-    // don't pack to 32b for Hopper
-    int vecSize = isHopper ? 1 : 32 / canonBits;
+    // Hopper may not contain 32b along kWidth; Ampere always does
+    int kBits = 8 * elemBytes * kWidth;
+    assert(kBits == 32 || isHopper);
+    int vecSize = kBits / canonBits;
```

Comment:
for non-transposed case, `vecSize = kBits / canonBits` was always 1, so the previous logic `(isHopper ? 1 : ...)` worked.

for transposed case, `vecSize` can be more than 1, since we're loading each value seaprately (i.e. `canonBits = 8 * elemBytes`).